### ISSUE TITLE
Improve logging for gossip messages

### ISSUE THREAD

**Issue by nazarhussain** [ChainSafe/lodestar#6554]
### Problem description

We recently encountered a scenario where we faced an error from the beacon API endpoint. 

```
error: Failed to send out exit message "PublishError.InsufficientPeers"
```

During the debugging, we could not find any such error been logged anywhere. The `validator-ejector` called the beacon API, which internally called the libp2p-gosshipsub, and that error was raised there and then triggered back to API response, and not logged any where in our code.

### Solution description

We should improve the logging for the gossip messages: 

1. Log all invocation any gossip message to any topic
2. Log success or error for any publishing message to any topic

### Additional context

Having those logs on the debug level would be sufficient.

**Comment by philknows** [comment]
Increasing priority on this to help debug issues where missed/orphaned blocks can yield better intel on the critical seconds of block production flow.

**Comment by nflaig** [comment]
> We should improve the logging for the gossip messages:
Log all invocation any gossip message to any topic
Log success or error for any publishing message to any topic

this is the case, not sure if this issue is still relevant but we always log publish errors as `error`

**Comment by philknows** [comment]
I believe this issue came about when we kept having exit message publishing issues with insufficient peers where our Lido validator-ejector instance was connected to and couldn't figure out why this was happening.

So we couldn't actually broadcast exit messages via Lodestar timely enough. @Faithtosin I believe we haven't investigated this further and are still not using Lodestar for this particular task?

**Comment by Faithtosin** [comment]
@philknows That's Correct! We switched to using Lighthouse for the mainnet! Hoodi is set up on Lodestar in case we need to test this.

**Comment by philknows** [comment]
@Faithtosin Are you able to try again with Lodestar v1.29.0 and validator-ejector there to see if we're still having the insufficient peer issue for exit message publishing? The problem with this issue is that I'm not sure if this is still required. And if it is, we need to be more specific about what we need to expose for further debugging.
